browser/BUILD.gn

```diff
@@ -452,6 +452,8 @@ source_set("stats_updater") {
   defines = [ "BRAVE_STATS_API_KEY=\"$brave_stats_api_key\"" ]
 
   sources = [
+    "brave_stats_tab_helper.cc",
```

Comment:
can we move this stuff into browser/brave_stats/BUILD.gn (and move the files as well)? This BUILD.gn is already big and cumbersome so it would be good to start moving things out when we touch them